Sources/PubNubChatComponents/Views/Reaction/MessageReactionListView.swift

```diff
@@ -0,0 +1,191 @@
+    self.reactionProvider = provider
+    
+    super.init(frame: .zero)
+    setupSubviews()
+  }
+  
+  public required init?(coder: NSCoder) {
+    self.reactionProvider = DefaultReactionProvider()
+    
+    super.init(coder: coder)
+    setupSubviews()
+  }
+  
+  lazy public var stackViewContainer = UIStackContainerView()
+
+  var allReactions: [String] {
+    reactionProvider.reactions
+  }
+  
+  open func setupSubviews() {
+    translatesAutoresizingMaskIntoConstraints = false
+    
+    stackViewContainer.stackView.alignment = .leading
+    stackViewContainer.stackView.spacing = 5.0
+    stackViewContainer.stackView.axis = .horizontal
+    
+    addSubview(stackViewContainer)
+    stackViewContainer.translatesAutoresizingMaskIntoConstraints = false
+    stackViewContainer.leadingAnchor.constraint(equalTo: leadingAnchor).isActive = true
+    stackViewContainer.topAnchor.constraint(equalTo: topAnchor).isActive = true
+    
+    stackViewContainer.trailingAnchor
+      .constraint(equalTo: trailingAnchor)
+      .priority(.overrideRequire).isActive = true
+    stackViewContainer.bottomAnchor
+      .constraint(equalTo: bottomAnchor)
+      .priority(.overrideRequire).isActive = true
+  }
+  
+  open func configure<Message>(
+    _ messageActionButtons: [MessageReactionButtonComponent],
+    message: Message,
+    currentUserId: String,
+    reactionProvider: ReactionProvider,
+    onMessageActionTap: ((MessageReactionButtonComponent?, Message, (() -> Void)?) -> Void)?
+  ) where Message : ManagedMessageViewModel {
+    
+    self.reactionProvider = reactionProvider
+    
+    let reactions = message.messageActions
+      .filter { $0.sourceType == "reaction" }
+      .map { (reaction: $0.value, userID: $0.pubnubUserId) }
+    
+    updateWith(reactions: reactions.map { $0.reaction } )
+    
+    reactionButtons.forEach { button in
+      button.isSelected = reactions.contains(where: { $0.userID == currentUserId && $0.reaction == button.reaction })
+    }
+    
+    reactionButtons.forEach { button in
+      button.didTap({ [weak message] button in
+        guard let message = message else { return }
+        
+        button?.isEnabled = false
+        onMessageActionTap?(button, message) { [weak button] in
+          DispatchQueue.main.async {
+            button?.isEnabled = true
+          }
+        }
+      })
+      .store(in: &button.externalCancellables)
+    }
+  }
+
+  open func configure<Message>(
+    _ message: Message,
+    currentUserId: String,
+    reactionProvider: ReactionProvider,
+    onMessageActionTap: ((MessageReactionButtonComponent?, Message, (() -> Void)?) -> Void)?
+  ) where Message : ManagedMessageViewModel {
+    configure(
+      reactionButtons,
+      message: message,
+      currentUserId: currentUserId,
+      reactionProvider: reactionProvider,
+      onMessageActionTap: onMessageActionTap
+    )
+  }
+  
+  func buttonFor(_ reaction: String) -> MessageReactionButtonComponent {
```

Comment:
Small nitpick, you can use an argument label before parameter name. But it's also OK
```
func button(for reaction: String) -> MessageReactionButtonComponent
```

Reply:
nice catch, your proposition is more swiftlike :) I will change it!